### ISSUE TITLE
"Either the username or the repository does not exist"

### ISSUE THREAD

**Issue by OverLordGoldDragon** [mandeep/Travis-Encrypt#24]
`travis-encrypt OverLordGoldDragon see-rnn`; [repository](https://github.com/OverLordGoldDragon/see-rnn); `travis-encrypt` version: 1.3.1, Win OS. Seems [cli.py](https://github.com/mandeep/Travis-Encrypt/blob/master/travis/cli.py#L66) looks [here](https://api.travis-ci.org/repos/OverLordGoldDragon/see-rnn), which shows build passing - but an error is thrown anyway; full trace below.

Any resolution?

<hr>

```
Traceback (most recent call last):
  File "d:\anaconda\envs\viz\lib\site-packages\travis\encrypt.py", line 49, in retrieve_public_key
    return response.json()['key'].replace(' RSA ', ' ')
KeyError: 'key'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\anaconda\envs\viz\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "d:\anaconda\envs\viz\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\Anaconda\envs\viz\Scripts\travis-encrypt.exe\__main__.py", line 7, in <module>
  File "d:\anaconda\envs\viz\lib\site-packages\click\core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "d:\anaconda\envs\viz\lib\site-packages\click\core.py", line 717, in main
    rv = self.invoke(ctx)
  File "d:\anaconda\envs\viz\lib\site-packages\click\core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "d:\anaconda\envs\viz\lib\site-packages\click\core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "d:\anaconda\envs\viz\lib\site-packages\travis\cli.py", line 68, in cli
    key = retrieve_public_key('{}/{}' .format(username, repository), url)
  File "d:\anaconda\envs\viz\lib\site-packages\travis\encrypt.py", line 52, in retrieve_public_key
    raise InvalidCredentialsError("Either the username: '{}' or the repository: '{}' does not exist. Please enter a valid username or repository name. The username and repository name are both case sensitive." .format(username, repository))
travis.encrypt.InvalidCredentialsError: Either the username: 'OverLordGoldDragon' or the repository: 'see-rnn' does not exist. Please enter a valid username or repository name. The username and repository name are both case sensitive.
```

**Comment by mandeep** [comment]
Hi @OverLordGoldDragon, thanks for the report. I'm able to replicate your problem using your username and repository name. However, when I dig deeper it seems that the Travis-CI endpoint is returning a 404 response for your repositories. 

When I run a simple request, I receive the following output even though the link shows Build Passing:
```python
>>> import requests
>>> requests.get("https://api.travis-ci.org/repos/OverLordGoldDragon/see-rnn")
<Response [404]>
```

I'll see if I can find something in the API docs that will let me access this information in another way.

**Comment by OverLordGoldDragon** [comment]
@mandeep Thanks; to notify, I worked around it by installing the [travis client](https://github.com/travis-ci/travis.rb#windows), and running `travis encrypt token-name --add deploy.password --com`. A secure encryption was generated, so my problem appears solved - though I'll leave closing to you if deciding to fix what I reported for `travis-encrypt`.

**Comment by mandeep** [comment]
Thanks, I wonder if this is an authentication issue. Does the official travis client work if you don't supply it with a token?

**Comment by OverLordGoldDragon** [comment]
@mandeep Like this? `travis encrypt --add deploy.password --com` - then I get:

```
Reading from stdin, press Ctrl+D when done
```
Pressing Ctrl+D (or anything else) just types `^D` into terminal, and other keys type directly also. When including `token-name` and running while `cd`'d into local repository, it automatically encrypts (after I approve it with `y` on prompt).

**Comment by mandeep** [comment]
I've found some repositories require an access token in order to generate a public key. In my case, it was a private repository that is now public.

I've pushed a fix for this in 114bc67f4431915d662ce4ba688ad02cdcff3da6. You can now use the following command which behaves the similar to your command:

`travis-encrypt --token TOKEN --private username repository`
